proposals/20220916-rules-and-plugin-distribution.md

```diff
@@ -0,0 +1,106 @@
+
+Also, we want to allow repository maintainers and advanced users to:
+
+* **Manage their repositories** so they can upload versioned rulesets and plugins in a public or private repository that they own
+
+Moreover, the intent is to create a future-proof implementation (it must be enough generic to support other kinds of **artifacts** in the future).
+
+## Non-Goals (out of scope)
+
+This effort currently **does not aim** to provide automated ways to:
+* Manage a Falco installation, in terms of installing/uninstalling/upgrading/starting/stopping the Falco binary or process
+* Modify the state of an existing and running Falco installation such as remotely adding/removing/upgrading plugins on a target cluster
+
+## Implementation
+
+We're going to use `falcoctl` as the helper tool that every user or script interacts with to achieve the goals above.
+
+Artifacts are **plugin**s or **rules file**s.
+
+The artifacts can be stored in any OCI compliant registry (they work like container image registries) such as GitHub packages. Registries can store artifacts for different platforms in the same way container images are distributed for different platforms (e.g. both linux+x86_64, linux+arm64 ...)
+
+Since registries can hold repos (e.g. `ghcr.io/falcosecurity/awesome-plugins`) with tags but do not have features to get a list of the repos we are going to use **index files** served over HTTPS to make it easier to locate the repo. Both registries and indexes can be public or private, using OCI repository login and http basic auth respectively.
+
+Users can supply a list of registries that can be used.
+
+### Index file overview
+
+The index will be implemented as a YAML file containing a list of entries. Each entry represents an artifact, and its structure should be like the following:
+
+```yaml
+  - name: k8saudit # mandatory
+    registry: ghcr.io # mandatory
+    repository: falcosecurity/k8saudit # mandatory****
+    type: plugin # # mandatory, can be (plugin|rulesfile) (e.g. `application/vnd.cncf.falco.*<type>*.layer.v1+tar.gz`)
+    description: Read Kubernetes Audit Events and monitor Kubernetes Clusters from EKS # mandatory because of the "search"
+    license: Apache-2.0 # License IDs refer to the SPDX License List at https://spdx.org/licenses
+    keywords: # mandatory because of the "search"
+      - monitoring
+      - security
+      - alerting
+      - metric
+      - troubleshooting
+      - run-time
+    home: https://falco.org
+    sources:
+      - https://github.com/falcosecurity/falco
+    maintainers:
+      - name: The Falco Authors
+        email: cncf-falco-dev@lists.cncf.io  
+```
+
+Notes:
+ - Artifact's releases and versions metadata are stored in the related artifact repository.
+ - References to artifacts are similar to references to a container image (see the section below).
+
+### Reference to an artifact
+
+It can be one of:
+- name (es. `k8saudit`) -> need to use the index for lookup
```

Comment:
Should we allow to specify a tag with this or does this automatically imply ":latest" like the one below?